waku/v2/protocol/relay/validators.go

```diff
@@ -13,19 +15,46 @@ import (
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/waku-org/go-waku/waku/v2/hash"
 	"github.com/waku-org/go-waku/waku/v2/protocol/pb"
+	"github.com/waku-org/go-waku/waku/v2/timesource"
 	"go.uber.org/zap"
 	proto "google.golang.org/protobuf/proto"
 )
 
 // Application level message hash
 func MsgHash(pubSubTopic string, msg *pb.WakuMessage) []byte {
-	// TODO: Other fields?
-	return hash.SHA256([]byte(pubSubTopic), msg.Payload, []byte(msg.ContentTopic))
+	timestampBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(timestampBytes, uint64(msg.Timestamp))
```

Comment:
the test vector is working fine, but `Timestamp` is a `int64` not `uint64`.

Reply:
well i guess it doesnt matter as long as the number is possitive (which would be). guess that a positive uint64 is represented in bytes just like an int64.

Reply:
`PutUint64` expects an `uint64` hence why I cast the timestamp to that type